src/Compilers/CSharp/Portable/Symbols/MemberSymbolExtensions.cs

```diff
@@ -397,7 +397,7 @@ internal static bool IncludeFieldInitializersInBody(this MethodSymbol methodSymb
             return methodSymbol.IsConstructor()
                 && !methodSymbol.HasThisConstructorInitializer()
                 && !(methodSymbol is SynthesizedRecordCopyCtor) // A record copy constructor is special, regular initializers are not supposed to be executed by it.
-                && !Binder.IsUserDefinedRecordCopyConstructor(methodSymbol);
+                && (!Binder.IsUserDefinedRecordCopyConstructor(methodSymbol) || methodSymbol is SynthesizedRecordConstructor);
```

Comment:
>IsUserDefinedRecordCopyConstructor [](start = 28, length = 34)

I think ```IsUserDefinedRecordCopyConstructor``` should be adjusted to not confuse primary constructor with a copy constructor.